### PR TITLE
feat: `/symbolicate`

### PR DESCRIPTION
**Pull request by szymonrybczak**

### CODE REVIEW

File: packages/vxrn/src/create.ts [hunk 1]
```diff
@@ -518,6 +519,34 @@ export const create = async (optionsIn: VXRNConfig) => {
...
+        const platform = inferPlatformFromStack(stack)
+
+        if (!platform) {
+          return new Response('Cannot infer platform from stack trace', { status: 400 })
+        }
+
+        // console.info({ msg: 'Starting symbolication', platform, stack })
+        const results = await processStacks(stack)
+        // console.info(results)
+        return new Response(JSON.stringify(results))
```
Comment (Aslemammad) [inline packages/vxrn/src/create.ts:542]: You can just `return results` as far i know, h3 will handle the response setting and other things.

File: packages/vxrn/src/create.ts [hunk 2]
```diff
@@ -518,6 +519,34 @@ export const create = async (optionsIn: VXRNConfig) => {
     defineEventHandler(() => `packager-status:running`)
   )
 
+  router.post(
+    '/symbolicate',
+    defineEventHandler(async (e) => {
+      const body = await readBody(e)
```
Comment (Aslemammad) [inline packages/vxrn/src/create.ts:525]: Doesn't readBody already return json most of the times? 

But if we're sure there are issues in this, we should just readRawBody utf8 and then try parsing it as we do now. So we do not need to do `body as string` anymore.

File: packages/vxrn/src/create.ts [hunk 3]
```diff
@@ -518,6 +519,34 @@ export const create = async (optionsIn: VXRNConfig) => {
...
+      // we cannot use JSON schema to validate the body.
+
+      try {
+        const { stack } = JSON.parse(body as string) as {
+          stack: ReactNativeStackFrame[]
+        }
+        const platform = inferPlatformFromStack(stack)
+
+        if (!platform) {
+          return new Response('Cannot infer platform from stack trace', { status: 400 })
```
Comment (Aslemammad) [inline packages/vxrn/src/create.ts:536]: Can't we use h3's `throw createError`!

File: packages/vxrn/src/create.ts [hunk 4]
```diff
@@ -518,6 +519,34 @@ export const create = async (optionsIn: VXRNConfig) => {
...
+        }
+
+        // console.info({ msg: 'Starting symbolication', platform, stack })
+        const results = await processStacks(stack)
+        // console.info(results)
+        return new Response(JSON.stringify(results))
+      } catch (error) {
+        console.log(error)
```
Comment (Aslemammad) [inline packages/vxrn/src/create.ts:544]: `throw createError`

File: packages/vxrn/src/nativePlugin.ts [hunk 5]
```diff
@@ -164,8 +164,12 @@ export function nativePlugin(options: {
                   return out
                 })
                 .join(';')
+              const magicString = new MagicString(code + '\n' + forceExports)
```
Comment (Aslemammad) [inline packages/vxrn/src/nativePlugin.ts:167]: Let's first create the magic string for code, and then inject the forceExports with a new line explicitly with the magic string instance itself, so we generate proper source maps.

**Review by Aslemammad — CHANGES_REQUESTED** [review]

File: packages/vxrn/src/create.ts (hunk 1 above)
Comment (szymonrybczak) [reply packages/vxrn/src/create.ts:542]: Oh, cool!

File: packages/vxrn/src/nativePlugin.ts (hunk 5 above)
Comment (szymonrybczak) [reply packages/vxrn/src/nativePlugin.ts:167]: good call!

File: packages/vxrn/src/create.ts (hunk 2 above)
Comment (szymonrybczak) [reply packages/vxrn/src/create.ts:525]: It returns JSON only when `content-type` is `application/json`, unfortunately React Native is sending this as `text/plain`. So I'll go with `readRawBody()` 👍 

Source: https://github.com/unjs/h3/blob/c04c458810e34eb15c1647e1369e7d7ef19f567d/src/utils/body.ts#L157

Comment (Aslemammad) [reply packages/vxrn/src/create.ts:525]: Yea, readRawBody makes more sense.

**Pull request closed without merging**